core/templates/pages/contributor-dashboard-admin-page/services/contributor-dashboard-admin-backend-api.service.spec.ts

```diff
@@ -0,0 +1,351 @@
+import { CsrfTokenService } from 'services/csrf-token.service';
+
+describe('Contributor dashboard admin backend api service', () => {
+  let cdabas: ContributorDashboardAdminBackendApiService;
+  let httpTestingController: HttpTestingController;
+  let csrfService: CsrfTokenService = null;
+  let successHandler = null;
+  let failHandler = null;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
+
+    cdabas = TestBed.get(ContributorDashboardAdminBackendApiService);
+    httpTestingController = TestBed.get(HttpTestingController);
+    csrfService = TestBed.get(CsrfTokenService);
+    successHandler = jasmine.createSpy('success');
+    failHandler = jasmine.createSpy('fail');
+
+    spyOn(csrfService, 'getTokenAsync').and.callFake(async() => {
+      return Promise.resolve('sample-csrf-token');
+    });
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  it('should add contribution rights to the user given the' +
+    'name when calling addContributionReviewerAsync', fakeAsync(() => {
+    let category = 'voiceover';
+    let languageCode = 'en';
+    let username = 'validUser';
+    let payload = {
+      category: category,
+      username: username,
+      language_code: languageCode
+    };
+    cdabas.addContributionReviewerAsync(
+      category, username, languageCode
+    ).then(successHandler, failHandler);
+
+    let req = httpTestingController.expectOne(
+      '/addcontributionrightshandler');
+    expect(req.request.method).toEqual('POST');
+    expect(req.request.body).toEqual(payload);
+
+    req.flush(
+      { status: 200, statusText: 'Success.'});
+    flushMicrotasks();
+
+    expect(successHandler).toHaveBeenCalled();
+    expect(failHandler).not.toHaveBeenCalled();
+  }
+  ));
+
+  it('should fail to add contribution rights to the user when user does' +
+    'not exists when calling addContributionReviewerAsync', fakeAsync(() => {
+    let category = 'voiceover';
+    let languageCode = 'en';
+    let username = 'InvalidUser';
+    let payload = {
+      category: category,
+      username: username,
+      language_code: languageCode
+    };
+    cdabas.addContributionReviewerAsync(
+      category, username, languageCode
+    ).then(successHandler, failHandler);
+
+    let req = httpTestingController.expectOne(
+      '/addcontributionrightshandler');
+    expect(req.request.method).toEqual('POST');
+    expect(req.request.body).toEqual(payload);
+    req.flush(
+      { error: 'User with given username does not exist'},
+      { status: 500, statusText: 'Internal Server Error'});
+    flushMicrotasks();
+
+    expect(successHandler).not.toHaveBeenCalled();
+    expect(failHandler).toHaveBeenCalledWith(
+      'User with given username does not exist');
+  }));
+
+  it('should get the data of contribution rights given the role' +
```

Comment:
Can we follow the test structure mentioned in https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests (bullet 2)? Grouping the lines together by test phase makes the tests a lot easier to read.